store/sqlstore/post_store.go

```diff
@@ -149,19 +225,45 @@ func (s *SqlPostStore) Update(newPost *model.Post, oldPost *model.Post) (*model.
 	return newPost, nil
 }
 
-func (s *SqlPostStore) Overwrite(post *model.Post) (*model.Post, *model.AppError) {
-	post.UpdateAt = model.GetMillis()
-
+func (s *SqlPostStore) OverwriteMultiple(posts []*model.Post) ([]*model.Post, *model.AppError) {
+	updateAt := model.GetMillis()
 	maxPostSize := s.GetMaxPostSize()
-	if appErr := post.IsValid(maxPostSize); appErr != nil {
-		return nil, appErr
+	for _, post := range posts {
+		post.UpdateAt = updateAt
+		if appErr := post.IsValid(maxPostSize); appErr != nil {
+			return nil, appErr
+		}
+	}
+
+	tx, err := s.GetMaster().Begin()
+	if err != nil {
+		return nil, model.NewAppError("SqlPostStore.Overwrite", "store.sql_post.overwrite.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
+	for _, post := range posts {
+		if _, err = tx.Update(post); err != nil {
```

Comment:
We can combine then all in one query, but as in the previous example, we have to follow the `CASE` approach, because the data changes from one case to another.

Reply:
Yes, that's what I meant. Use the CASE approach.